src/Tizen.Network.Stc/Tizen.Network.Stc/StcRule.cs

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright (c) 2018 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+
+namespace Tizen.Network.Stc
+{
+    /// <summary>
+    /// A class for managing the Stc Rules to match applications.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public class StcRule : IDisposable
+    {
+        internal IntPtr _ruleHandle = IntPtr.Zero;
+        internal bool _disposed;
+
+        /// <summary>
+        /// Creates a StcRule object.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown when the Stc is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to an invalid operation.</exception>
+        public StcRule()
+        {
+            Log.Debug(Globals.LogTag, "New StcRule.");
+            int ret = Interop.Stc.Rule.Create(StcManagerImpl.Instance.GetSafeHandle(), out _ruleHandle);
+            if(ret != (int)StcError.None)
+            {
+                Log.Error(Globals.LogTag, "Failed to create Rule handle, Error - " + (StcError)ret);
+                StcErrorFactory.ThrowStcException(ret);
+            }
+        }
+
+        /// <summary>
+        /// Destroy the StcRule object
+        /// </summary>
+        ~StcRule()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// A method to destroy the managed StcRule objects.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            int ret = Interop.Stc.Rule.Destroy(_ruleHandle);
+            if (ret != (int)StcError.None)
+            {
+                Log.Error(Globals.LogTag, "Failed to destroy Rule handle, Error - " + (StcError)ret);
+                StcErrorFactory.ThrowStcException(ret);
```

Comment:
Do not throw exception on Dispose method

Reply:
Done.